scripts/verify_golangci-lint_version.sh

```diff
@@ -27,7 +27,13 @@ if [ -z "$GOLANGCI_LINT_PRESENT" ]; then
     install_golangci_lint
     exit 0
 fi
-GOLANGCI_LINT_INSTALLED=v$(golangci-lint version | grep -oP 'version \K[0-9.]+')
+GOLANGCI_LINT_INSTALLED=v$(golangci-lint version | grep -oP 'version \K[0-9.]+' 2>&1)
+
+# Check if the result contains "invalid option" which may happen on macOS with outdated grep
+if [[ "$GOLANGCI_LINT_INSTALLED" == *"invalid option"* ]]; then
+    echo "If you are on Mac, run: brew install grep."
+    exit 1
+fi
```

Comment:
thx for the PR. Since MacOS is widely used, can we support it so that contributors can run verification (including golangci-lint) locally on their macOS?